pkg/apis/wsecurity/v1/README.md

```diff
@@ -0,0 +1,47 @@
+# Security Data Package
+
+This package serves as the beating heart of Guard.
+
+It defines data structures that meet the [v1 interface](v1.go).
+
+## The wsecurity/v1 API
+
+The [v1 interface](v1.go) includes three interfaces:
+
+- Profile: describing a sample of a data type
+- Pile: accumulating multiple samples of a data type to enable learning.
+- Config: the rules describing what is expected from the data type.
+
+## Core Activity
+
+Per Sample:
+
+1. Profile.Profile(...sample...) - Create a profile from the sample
+1. Config.Decide(profile)  - Decide if it conforms to the config rules
+1. Pile.Add(profile)  - Add it to a pile.
+
+Periodically:
+
+1. Pile.Merge(someOtherPile) - Merge someOtherPile to Pile. Note that Piles consume other piles - once someOtherPile is added to pile it is no longer usable.
+1. Config.Learn(pile) - Learn a new config rules based on a pile.
```

Comment:
Does this consume the `Pile`, or no? Given the previous `Merge` operation, it seems worth specifying.

Reply:
When I wrote 'consume' I meant to say that some of the data in someOtherPile may now be reused by pile, to avoid deep copy when possible.

Reply:
"Learn a new config rules based on a pile" doesn't say whether it consumes the pile or not.

Reply:
(tnx, I originally thought the language and using 'consume'  was unclear..   will be fixed)